packages/push-platform-notifications-controller/src/services/services.ts

```diff
@@ -0,0 +1,339 @@
+import type { FirebaseApp } from '@firebase/app';
+import { getApp, initializeApp } from '@firebase/app';
+import { getToken, deleteToken } from '@firebase/messaging';
+import { getMessaging, onBackgroundMessage } from '@firebase/messaging/sw';
+import type {
+  Messaging,
+  MessagePayload,
+  Unsubscribe,
+} from '@firebase/messaging/sw';
+import type { Notification as INotification } from '@metamask/notifications-controller';
+import log from 'loglevel';
+
+import {
+  FIREBASE_API_KEY,
+  FIREBASE_APP_ID,
+  FIREBASE_AUTH_DOMAIN,
+  FIREBASE_MEASUREMENT_ID,
+  FIREBASE_MESSAGING_SENDER_ID,
+  FIREBASE_PROJECT_ID,
+  FIREBASE_STORAGE_BUCKET,
+  PUSH_NOTIFICATIONS_SERVICE_URL,
+  VAPID_KEY,
+} from '../constants';
+import {
+  onNotificationClick,
+  onPushNotification,
+} from '../utils/get-notification-message';
+
+const url = PUSH_NOTIFICATIONS_SERVICE_URL;
+const REGISTRATION_TOKENS_ENDPOINT = `${url}/v1/link`;
+const sw = self as unknown as ServiceWorkerGlobalScope;
```

Comment:
Is this controller only intended to be used in a service worker? If so, I'm curious whether that should be noted somewhere in the README. We don't have any precedent to this, but some kind of blurb after the description might be helpful.

Reply:
Added "This is mainly focused on wrapping and delivering a common interface over firebase related interactions for push notifications clients [mobile|extension|portfolio]." to README, for clarity.